spark/src/main/scala/org/apache/spark/sql/comet/CometNativeScanExec.scala

```diff
@@ -53,29 +55,50 @@ case class CometNativeScanExec(
     disableBucketedScan: Boolean = false,
     originalPlan: FileSourceScanExec,
     override val serializedPlanOpt: SerializedPlan)
-    extends CometLeafExec {
+    extends CometLeafExec
+    with DataSourceScanExec {
```

Comment:
IIRC `DataSourceScanExec` is for V1.
Is this ok for future V2 support?

Reply:
I don't know enough about the APIs to say for sure, but it was my understanding that V1 -> `native_datatusion` reader and V2 -> `native_iceberg_compat`. Maybe @parthchandra can clarify.

Reply:
V2 sources will go thru BatchScanExec and at some point native_datafusion can be extended to V2 by implementing a NativeBatchScanExec the same way Spark extended V1 parquet to V2; the V2 path is a wrapper around V1. Mostly, people reading Parquet do not bother going via the V2 APIs so it is not urgent for us to support V2 in native_datafusion. 
iceberg_compat is basically a bunch of lower level APIs that the iceberg integration will call (eventually). In the meantime, I added the calls to these APIs in the V1 path so we can test the APIs end to end.